src/open_r1/utils/routed_morph.py

```diff
@@ -0,0 +1,117 @@
+# coding=utf-8
+# Copyright 2025 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+from typing import List, Optional
+
+
+class RoutedMorphSandbox:
+    """
+    Client for the MorphCloud router service that mimics the API of MorphCloud's Sandbox.
+    
+    This class provides a simple interface to execute code via a central MorphCloud router,
+    which manages sandbox creation and cleanup. It allows batch processing of multiple scripts
+    in a single request for improved efficiency.
+    
+    Attributes:
+        router_url (str): The URL of the MorphCloud router service.
+        timeout (int): Execution timeout in seconds.
+        request_timeout (int): HTTP request timeout in seconds.
+    """
+    
+    def __init__(self, router_url: str, timeout: int = 30, request_timeout: int = 60):
+        """
+        Initialize the routed MorphCloud sandbox client.
+        
+        Args:
+            router_url: The URL of the MorphCloud router, including host and port.
+            timeout: Default execution timeout in seconds.
+            request_timeout: Default HTTP request timeout in seconds.
+        """
+        self.router_url = router_url
+        self.timeout = timeout
+        self.request_timeout = request_timeout
+    
+    def run_code(self, scripts: List[str], 
+                language: str = "python", 
```

Comment:
As mentioned above, perhaps we can relax the language here and have a list of languages?